utils/appliance.py

```diff
@@ -444,8 +444,12 @@ def rest_api(self):
     @lazycache
     def address(self):
         # If address wasn't set in __init__, use the hostname from base_url
-        parsed_url = urlparse(self.url or store.base_url)
-        return parsed_url.netloc
+        if getattr(self, "_url", None) is not None:
+            parsed_url = urlparse(self.url)
+            return parsed_url.netloc
+        else:
+            parsed_url = urlparse(store.base_url)
+            return parsed_url.netloc
```

Comment:
pytest_store can look up the base_url from env, or if a current_appliance is set, then it asks current_appliance for url. The url looks for self.address which by default used to refer to store.base_url. I addded the or there but accessing hte url makes it go to infinite recursion. This fixes the issue by not triggering the url if we won't use it.